### PR TITLE
feat: add top-level query connection decorator

### PR DESCRIPTION
**Pull request by StevenLangbroek**
Hey @rogerballard!

Had a use-case for this, and thought it'd be easier to open a PR to discuss instead of just opening an issue. Would love to hear your thoughts. Did I miss anything, either test or code-wise?

Top-level connections might seem counterintuitive, but in essence they're a connection between the viewer and the entities you're querying.

Thanks!

### CODE REVIEW

File: test/starwars-app/faction.resolver.ts [hunk 1]
```diff
@@ -27,6 +28,14 @@ export class FactionResolver extends GlobalIdFieldResolver(Faction) {
     return this.factionService.getEmpire();
   }
 
+  @QueryConnection(() => Faction)
+  factions(@Args() args: ConnectionArgs) {
+    return connectionFromArray(
+      [this.factionService.getRebels(), this.factionService.getEmpire()],
```
Comment (StevenLangbroek) [inline test/starwars-app/faction.resolver.ts:34]: I _should_ probably add a `getFactions()` method to the factionService... What do you think? Also query a nested connection to make sure everything checks out?

**Comment by StevenLangbroek** [comment]
Question about [this line](https://github.com/rogerballard/nestjs-relay/pull/169/files#diff-2d5a46a38d4a669e90df4b9867e6fa7144173bd43c9ea491a65149304c04eb0aR15): in all of our code and in the NestJS docs this should be `@Resolver(() => Faction)` instead of `@Resolver(Faction)`. Is there any difference in behavior?

**Review by ThisIsMissEm — APPROVED** [review]
LGTM

**Comment by rogerballard** [comment]
Hey @StevenLangbroek and @ThisIsMissEm 👋 first of all I'd like to thank you for your feature submission!

I have had a read through and it looks great at first glance. I'll do a more in-depth review over the weekend and contribute some improvements to the docs to go with the new feature.

On another note: I'm glad that you are finding this library useful, and I'm interested in any other feedback you may have (perhaps via email or in a separate issue).

**Comment by StevenLangbroek** [comment]
And thank you for doing all the heavy lifting here :). I have some feedback based on local development experience, what's a good place to discuss.

**Comment by rogerballard** [comment]
> And thank you for doing all the heavy lifting here :). I have some feedback based on local development experience, what's a good place to discuss?

Please add your feedback into a new issue so that we can discuss separate to this PR 👍

**Comment by StevenLangbroek** [comment]
@rogerballard any chance you have semantic-release set up to do prereleases? If so I'd like to give this change a go in an actual project I'm implementing in relay spec.

File: test/connection.e2e-spec.ts [hunk 2]
```diff
@@ -172,6 +172,91 @@ describe('Connection', () => {
...
+            }
+          }
+        }
+      }
+    }
+  `;
+
+    const response = await request(app.getHttpServer()).post('/graphql').send({ query });
+
+    console.log(JSON.stringify(response.body, null, 2));
```
Comment (rogerballard) [inline test/connection.e2e-spec.ts:199]:
```suggestion
```

File: test/starwars-app/faction.service.ts [hunk 3]
```diff
@@ -53,6 +53,10 @@ export class FactionService {
     };
   }
 
+  public getFactions() {
```
Comment (rogerballard) [inline test/starwars-app/faction.service.ts:56]: :+1:

**Review by rogerballard — APPROVED** [review]

**Pull request merged**